scripts/product_migration.py

```diff
@@ -0,0 +1,255 @@
+import pysolr
+import boto
+from boto.s3.connection import S3Connection
+
+CAPTURE_STREAM_L0_REGEX = "^[0-9]{10}[-_].*l0$"
+CAPTURE_STREAM_L1_REGEX = "^[0-9]{10}[-_].*l1-flags$"
+CAPTURE_STREAM_L1_REGEX_HISTORIC = "^[0-9]{10}[-_].*l1_flags$"
+
+TELESCOPE = "MeerKATTelescopeProduct"
+
+GENERIC_PLACEHOLDER = "PlaceholderProduct"
+PRODUCT_PLACEHOLDER = "MeerKATVisibilityProduct-Placeholder"
+FLAGS_PLACEHOLDER = "MeerKATFlagProduct-Placeholder"
+DEFUNCT_STATE = 'DEFUNCT'
+
+PRODUCT_REDUCTON_OLD = "MeerKATTelescopeProduct"
+PRODUCT_NEW = "MeerKATVisibilityProduct"
+FLAGS_NEW = "MeerKATFlagProduct"
+
+
+def new_product_exists(product, solr_conn):
+    block_id = product['CaptureBlockId']
+    results = solr_conn.search(
+        f'CaptureBlockId:{block_id} && '
+        f'CAS.ProductTypeName:{PRODUCT_NEW}')
+    return results.hits
+
+
+def get_flag_placeholder_product(product, solr_conn):
+    block_id = product['CaptureBlockId']
+    results = solr_conn.search(
+        f'CaptureBlockId:{block_id} && '
+        f'CAS.ProductTypeName:{GENERIC_PLACEHOLDER} && '
+        f'ReductionName:{FLAGS_NEW}'
+    )
+    return results
+
+
+def get_product_placeholder_product(product, solr_conn):
+    block_id = product['CaptureBlockId']
+    results = solr_conn.search(
+        f'CaptureBlockId:{block_id} && '
+        f'CAS.ProductTypeName:{GENERIC_PLACEHOLDER} && '
+        f'ReductionName:{PRODUCT_REDUCTON_OLD}'
+    )
+    return results
+
+
+def create_l0_product_from_placeholder(rdb_product, placeholder_product, solr, s3_conn, update_set):
+    product_id = f'{placeholder_product["Prefix"]}-sdp-l0-visibility'
```

Comment:
Might be worth just using CaptureBlockId rather than Prefix here. Although I believe they should be identical, you are more likely to find a product missing a Prefix than a CaptureBlockId.